bundles/org.openhab.binding.sonnen/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,137 @@
+			<parameter name="hostIP" type="text" required="true">
+				<context>network-address</context>
+				<label>IP Address</label>
+				<description>Please add the IP Address of your sonnen battery</description>
+			</parameter>
+			<parameter name="refreshInterval" type="integer" unit="s" min="0" max="1000">
+				<label>Refresh Interval</label>
+				<description>How often in seconds the sonnen battery should schedule a refresh after a channel is linked to an item.
+					Valid input is 0 - 1000</description>
+				<advanced>true</advanced>
+				<default>30</default>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<!-- sonnen Channel Type -->
+	<channel-type id="batteryCharging">
+		<item-type>Switch</item-type>
+		<label>Battery Charging</label>
+		<description>Indicates if the Battery is charging at that moment</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="batteryDischarging">
+		<item-type>Switch</item-type>
+		<label>Battery Discharging</label>
+		<description>Indicates if the Battery is discharging at that moment</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="batteryFeedIn">
+		<item-type>Number:Energy</item-type>
+		<label>Battery Feed In</label>
+		<description>Indicates the actual current feeding to the Battery in watt. Otherwise 0</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="batteryDispense">
+		<item-type>Number:Energy</item-type>
+		<label>Battery Dispense</label>
+		<description>Indicates the actual current dispense from the Battery in watt. Otherwise 0</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="consumption">
+		<item-type>Number:Energy</item-type>
+		<label>Consumption</label>
+		<description>Indicates the actual consumption of the consumer in watt</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="gridFeedIn">
+		<item-type>Number:Energy</item-type>
+		<label>Grid Feed In</label>
+		<description>Indicates the actual feeding to the Grid in watt. Otherwise 0</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="gridReceive">
+		<item-type>Number:Energy</item-type>
+		<label>Grid Receive</label>
+		<description>Indicates the receiving from the the Grid in watt. Otherwise 0</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="solarProduction">
+		<item-type>Number:Energy</item-type>
+		<label>Solar Production</label>
+		<description>Indicates the actual production of the Solar system in watt</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="batteryLevel">
+		<item-type>Number</item-type>
+		<label>Battery Level</label>
+		<description>Indicates the actual Battery Level in % from 0 - 100</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="flowConsumptionBattery">
+		<item-type>Switch</item-type>
+		<label>Flow Battery -> Consumption</label>
```

Comment:
What is recommended for such a label is 2 or 3 words with each word starting by a upper letter.
So that would better if you can avoid special characters like "->". So I thing "Flow Battery Consumption" is better.
Same comment for all your channels.